packages/components/src/border-control/test/index.js

```diff
@@ -43,7 +43,9 @@ const rerenderBorderControl = ( rerender, customProps ) => {
 
 const openPopover = () => {
 	const toggleButton = screen.getByLabelText( toggleLabelRegex );
-	fireEvent.click( toggleButton );
+	act( () => {
```

Comment:
These are a bit surprising to me. Isn't `fireEvent.click` supposed to already do `act` under the hood? cc @eps1lon

Reply:
Every `fireEvent.*` method wraps the `dispatchEvent` in `act`.  If `act` is required to flush some updates then we should investigate. I'll check this out later anyway since this is a nice big project to check Testing Library compatibility.

Reply:
Hi there! thanks for chiming in. I've looked a bit and it seems we're just being too conservative in most cases. There are some places where the test broke if I remove it (Example here https://github.com/WordPress/gutenberg/blob/9c7b2060734ef6faf41116c4fd42c259b5d53a58/packages/components/src/focal-point-picker/test/index.js#L109-L112)

I've elected not to remove the unnecessary ones right now. (this PR)

Reply:
Removed that specific `act` call and `npm run test-unit` worked fine (apart from the docker related errors. don't know how to fix them).

You'd still want to wrap e.g. `HTMLElement.focus` calls. But wrapping `fireEvent` calls is generally unnecessary. If there are other patterns I missed, please point the specific tests out that need `act` where it looks unnecessary.

Reply:
After some time I'm working on finishing the React 18 migration and I ran into this `act` issue again. The `BorderControl` test is failing without an extra `act`, explained here: #45241.

@eps1lon You might be interested why the automatic `act` inside `fireEvent` is not enough: we have an async callback running inside a `.then` handler and we need to call an async `await act( async () => {} )` to flush the state updates that happen inside the `.then` handler.